src/main/java/org/opentripplanner/updater/vehicle_positions/VehiclePositionPatternMatcher.java

```diff
@@ -107,11 +110,14 @@ public void applyVehiclePositionUpdates(List<VehiclePosition> vehiclePositions)
             return null;
         }
 
-        var instant = Optional.of(vehiclePosition.getTimestamp())
-                .filter(t -> t > 0)
-                .map(Instant::ofEpochSecond)
-                .orElse(Instant.now());
-        TripPattern pattern = getPatternForTrip.apply(trip, instant);
+        var localDate =
+                Optional.ofNullable(vehiclePosition.getTrip())
```

Comment:
´vehiclePosition.getTrip()´ is not null, as it is checked by `if (!vehiclePosition.hasTrip()) {`, and `start_date ` is required always, when `trip_id` is present

Reply:
I've made start_date required in https://github.com/opentripplanner/OpenTripPlanner/pull/3981/commits/b74b73927dc8f745b09100ccf76c66bd6da98d05